### ISSUE TITLE
Feature Request: Add Option to Store SSH Keys Directly in Tabby

### ISSUE THREAD

**Issue by usmanibrahim74** [Eugeny/tabby#10040]
It would be great to have an option in Tabby to store SSH keys directly within the application, similar to the functionality available in tools like Termius. This feature would allow users to sync SSH connections, including the actual key content, across multiple devices, making it easier to manage connections securely and efficiently.

**Feature Details:**

**Store SSH Keys Securely in Tabby:**
Currently, Tabby allows users to specify paths to SSH keys, but it would be highly convenient to store the actual SSH key files directly in the application.
This would enable users to sync their SSH configurations, including keys, across multiple systems. When Tabby is installed on a new device, all connections and keys could be loaded immediately, without needing to set up SSH keys manually or transfer them.

**Encryption and Security:**
Ensure that any stored SSH keys are encrypted within Tabby’s configuration file, using a master password or system keychain for decryption. This would protect users' sensitive keys even if the configuration file is accessed by unauthorized parties.
A master password prompt could be added to unlock keys when Tabby starts, ensuring no key is exposed in plain text.

**Cross-Platform Sync:**
Add an optional, secure way to sync Tabby configurations (including SSH keys) through a cloud provider like GitHub or a custom secure service, similar to how Termius syncs configurations.
Alternatively, allow users to export/import encrypted configurations easily for manual backup across devices.
Benefits:

**Convenience:** Users can set up Tabby on a new device with all their SSH keys and connections ready to use, streamlining the setup process.
Security: Encrypting keys within Tabby ensures they remain secure even if configurations are shared.
Consistency: Enables users to maintain a consistent setup across multiple devices without needing to manually transfer SSH key files.
**Conclusion:** Adding the ability to store SSH keys directly in Tabby, with encryption and sync support, would make Tabby a more versatile and competitive tool for managing SSH connections across multiple platforms.

Thank you for considering this feature! It would significantly enhance the user experience for those managing SSH connections on multiple devices.

**Comment by mangfu26** [comment]
Yes, we desperately need this feature, especially for ssh key synchronization, which currently only synchronizes the path to the ssh key but not the contents of the ssh key.

**Comment by lronkitty** [comment]
Yes!!!

**Comment by Eugeny** [comment]
Settings -> Vault

**Comment by atlantsecurity** [comment]
Settings - Vault opens an empty screen. It does not allow adding anything there.
